File: src/optic/lua/ssa/instructions/TableIndex.java
package optic.lua.ssa.instructions;

import optic.lua.ssa.*;

public final class TableIndex implements Step {
	private final Register table;
	private final Register key;
	private final Register out;

	public TableIndex(Register table, Register key, Register out) {
		this.table = table;
		this.key = key;
		this.out = out;
	}

	@Override
	public String toString() {
		return String.format("lookup %s = %s[%s]", out, table, key);
	}
}

File: src/optic/lua/ssa/instructions/TableIndex.kt
package optic.lua.ssa.instructions

import optic.lua.ssa.Register
import optic.lua.ssa.Step

class TableIndex(val table: Register, val key: Register, val out: Register) : Step {
    override fun toString(): String {
        return "lookup $out = $table[$key]"
    }
}
